emcc.py

```diff
@@ -37,24 +41,28 @@
 from enum import Enum
 from subprocess import PIPE
 
+from tools.response_file import substitute_response_files
+
+# read response files very early on so that all subsequent code that accesses
+# sys.argv will see the expanded content.
+try:
+  sys.argv = substitute_response_files(sys.argv)
+except IOError as e:
+  raise Exception('Unable to parse response files!\n' + str(e))
```

Comment:
I wonder if its worth moving this into `config.py` so it applies equally to all commands?

Reply:
And presumably if we do this here we should avoid doing it again down in the `run` function?

Reply:
> I wonder if its worth moving this into `config.py` so it applies equally to all commands?

Works for me. Moved it over.

> And presumably if we do this here we should avoid doing it again down in the `run` function?

Good catch, forgot to remove that. Updated.

Reply:
> > I wonder if its worth moving this into `config.py` so it applies equally to all commands?
> 
> Works for me. Moved it over.

This did break emar.py response file handling, now needs #13982 to lands first.